### PR TITLE
refactor: remove hooks

### PR DESCRIPTION
**Pull request by lightsing**
The hooks pattern seems to be outdated since we no longer doing post tx checking. The only reason to have hook is to reuse the rlp bytes, which we can do it in a trivial way -- returning it after handling the block.

This pr removes the ExecutionHook and changes the way we reusing the rlp bytes.

### CODE REVIEW

**Comment by matthiasgoergens** [comment]
Could you add in the PR description a bit about **why** we are doing that, and very high level overview of **what** we are doing?

The **what** is mostly to serve as an introduction to the reader.  The details of the 'what' will be answered in the diff itself.

File: crates/core/src/executor/mod.rs
Comment (matthiasgoergens) [inline crates/core/src/executor/mod.rs:145]: Why do we need to clone here?

**Review by matthiasgoergens — APPROVED** [review]
Code looks mostly alright, but please see my request for some extra context in the description.

Perhaps also post some benchmarks?  (I'm assuming we are doing this to help decrease the cycle count?)

I'm approving now, but please address all comments before merging.  Thanks!

File: crates/core/src/executor/mod.rs
Comment (lightsing) [reply crates/core/src/executor/mod.rs:145]: We need to calculate l1 fee which is related to rlp bytes, also in chunk mode, we need calc the keccak of all tx rlp bytes. So it's for reusing the rlp bytes, avoid encoding it twice.

**Comment by lightsing** [comment]
> Could you add in the PR description a bit about **why** we are doing that, and very high level overview of **what** we are doing?
> 
> The **what** is mostly to serve as an introduction to the reader. The details of the 'what' will be answered in the diff itself.

I will add it next time.

**Pull request merged**
